src/V3PreLex.l

```diff
@@ -112,6 +116,47 @@ bom             [\357\273\277]
 <INITIAL>"`undef"       { FL_FWDC; return VP_UNDEF; }
 <INITIAL>"`undefineall" { FL_FWDC; return VP_UNDEFINEALL; }
 <INITIAL>"`error"       { FL_FWDC; if (!pedantic()) return VP_ERROR; else return VP_DEFREF; }
+    /* We wanted this to be next to `protect But it must be before `pramga */
+    /* we win only because we both match to the end of the line so the length */
+    /* is equal and we are first*/
+<INITIAL>"`pragma"{wsn}+"protect"[^\n\r]*[\n\r] { FL_FWDC; linenoInc();
+                          if (v3Global.opt.preprocOnly()) {
+                              return VP_TEXT;
+                          }
+                          int res, yypos = 7; /* length of `pragma */
+                          char *yyt = yytext;
+                          yyt[yyleng] = 0; /* this is OK because the last char here is \n or \r */
+                          yyt += yypos;
+                          if ((res = sscanf(yyt, " protect %n", &yypos)) != 0)
+                            yyerrorf("scanf failure");
+                          yyt += yypos;
+                          if (bytes == 0) {
+                            res = sscanf(yyt, "encoding = (enctype = \"BASE64\", line_length = %d, bytes = %d)%n", &length, &bytes, &yypos);
+                            if (res == 2)
+                                return VP_TEXT; /* silently skip it */
+                            bytes = 0;
+                          } else if ((strncmp(yyt, "key_block", 9) == 0) || (strncmp(yyt, "data_block", 10) == 0)) {
+                            BEGIN(BASE64);
+                            return VP_TEXT;
+                          } else {
+                              return VP_TEXT;
+                          }
+                        }
```

Comment:
I think we discussed having a new warning saying the protected regions are being ignored?

Reply:
I'll take a look.